### PR TITLE
Support debug tool bar float on custom title bar

### PR DESCRIPTION
**Pull request by sandy081**
**Motivation**: My setup contains custom titlebar without command center and tabs hidden and I want to have debug tool bar on the custom title bar.

image

Floating debug tool bar is allowed to be placed anywhere in the two rows below the title bar. I enabled this to also be placed on the custom title bar. Please check the following gif.

Recording 2023-10-31 at 14 29 46

Also note that, I have shortened the tool bar when tabs and command center are hidden to match the titlebar height. In general I liked the shortened version. So, how about shortening the debug tool bar always?

CC @isidorn

### CODE REVIEW

**Comment by isidorn** [comment]
I like these changes overall 👏 

+1 for shortening the debug toolbar all the time

fyi @connor4312

File: src/vs/workbench/contrib/debug/browser/debugToolBar.ts [hunk 1]
```diff
@@ -230,12 +230,43 @@ export class DebugToolBar extends Themable implements IWorkbenchContribution {
...
+
+		this.setYCoordinate(y);
+	}
+
+	private setYCoordinate(y = this.yCoordinate): void {
+		const [yMin, yMax] = this.yRange;
+		y = Math.max(yMin, Math.min(y, yMax));
+		this.$el.style.top = `${y}px`;
+		this.yCoordinate = y;
+		this.storageService.store(DEBUG_TOOLBAR_Y_KEY, y, StorageScope.PROFILE, StorageTarget.MACHINE);
```
Comment (connor4312) [inline src/vs/workbench/contrib/debug/browser/debugToolBar.ts:242]: This is writing to the StorageServer on every mousemove event. Should this be debounced?

**Comment by connor4312** [comment]
nice! One comment

**Comment by sandy081** [comment]
@connor4312 Addressed the feedback

**Review by connor4312 — APPROVED** [review]

**Pull request merged**